src/main/java/jdwp/JDWPMethod.java

```diff
@@ -193,15 +195,22 @@ public static void write(LocalVariableImpl var, GDBControl gc, PacketStream answ
                     answer.writeStringOrEmpty(var.genericSignature());
                     answer.writeInt(var.getLength());
                     answer.writeInt(var.slot());
-                    JDWP.localsByID.put(var.slot(), var);
+
+                    // Add this frame and its associated locals to the map
+                    // This frame will be mapped to the current method and this command will not be called again
+                    // with next breakpoint hit.
+                    MIFrame frame = JDWP.framesById.get(0);
+                    Map<Integer, LocalVariableImpl> frameMap = JDWP.localsByFrame.get(frame);
+                    frameMap.put(var.slot(), var);
```

Comment:
where do we use this `frameMap`?

Reply:
Don't think this really updates `localsByFrame.get(frame)` since you're storing it into a new variable.

Reply:
ah just on line 204 since the code was getting long so I thought I'd break it down. I can reduce it to the same line though!

Reply:
You might need to add `JDWP.localsByFrame.put(frame, frameMap)` after 204

Reply:
hmm I think hashmap's `get` method returns a reference so 203-204 should do the same as 
```JDWP.localsByFrame.get(frame).put(var.slot(), var)```
wdyt?

Reply:
You are absolutely right!! Sorry (language confusion) :)